client/components/New.js

```diff
@@ -0,0 +1,177 @@
+
+const React = require('react');
+const SkillCategoryLabel = require('./SkillCategoryLabel');
+const RoleLabel = require('./RoleLabel');
+const Lego = require('./Lego');
+const ReactBootstrap=require('react-bootstrap')
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state={
+      error:'' ,
+      categories:[
+      {skillCategoryName:'Category Name',skills:['skill-1','skill-2','skill-3','skill-4','skill-5','skill-6'],categoryImg:'client/assets/icons/1.png'},
+      {skillCategoryName:'Category Name',skills:['skill-1','skill-2','skill-3'],categoryImg:''},
+      {skillCategoryName:'Category Name',skills:['skill-1','skill-2','skill-3'],categoryImg:''},
+      {skillCategoryName:'Category Name',skills:['skill-1','skill-2','skill-3'],categoryImg:''},
+      {skillCategoryName:'Category Name',skills:['skill-1','skill-2','skill-3'],categoryImg:''},
+      {skillCategoryName:'Category Name',skills:['skill-1','skill-2','skill-3','skill-3','skill-3','skill-3'],categoryImg:''}],
+      roleName:'role Name',lego:'',description:'bla bla bla bla bla bla bla bla bla bla bla bla bla bla ',
+      keywords: [{id:1,text:'keyword-1'},{id:1,text:'keyword-2'},{id:1,text:'keyword-3'}],
+      suggestions:["Banana", "Mango", "Pear", "Apricot"]
+    }
+    this.setRoleProps=this.setRoleProps.bind(this)
+    this.setCategoryDetails=this.setCategoryDetails.bind(this)
+    this.setLego=this.setLego.bind(this)
+    this.create=this.create.bind(this)
+  }
+  setRoleProps(roleName_,keywords_,description_){
+
+    this.setState({
+      roleName: roleName_, keywords: keywords_,description:description_
+    });
+
+  }
+  setLego(legoImg){this.setState({lego:legoImg})}
+
+  setCategoryDetails(id,categoryName,skills,iconUrl){
+    console.log('New.js',iconUrl)
+    let c=this.state.categories
+    c[id-1].skillCategoryName=categoryName;
+    c[id-1].skills=skills;
+    c[id-1].categoryImg=iconUrl;
+    this.setState({
+      categories: c
+    });
+  }
+
+  componentDidUpdate(){console.log(this.state.categories[0].categoryImg)}
+
+  create(){
+      console.log('posting .....')
+      const opts={}
+      opts.name=this.state.roleName
+      opts.description=this.state.description
+      opts.image=this.state.lego
+      opts.keywords=this.state.keywords.map((item,index)=>{return(item.text)})//['a','b','c']
+      opts.skillCategory=[]
+      opts.skillCategory=[1,2,3,4,5,6].map((item,index)=>{
```

Comment:
Here, you are also assuming there are always 6 skill categories. This should be parametric such as `skillCategory.length`